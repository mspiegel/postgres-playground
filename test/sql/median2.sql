CREATE TABLE intvals2(val int, color text);

-- Test empty table
SELECT median2(val) FROM intvals2;

-- Integers with odd number of values
INSERT INTO intvals2 VALUES
       (1, 'a'),
       (2, 'c'),
       (9, 'b'),
       (7, 'c'),
       (2, 'd'),
       (-3, 'd'),
       (2, 'e');

SELECT * FROM intvals2 ORDER BY val;
SELECT median2(val) FROM intvals2;

-- Integers with NULLs and even number of values
INSERT INTO intvals2 VALUES
       (99, 'a'),
       (NULL, 'a'),
       (NULL, 'e'),
       (NULL, 'b'),
       (7, 'c'),
       (0, 'd');

SELECT * FROM intvals2 ORDER BY val;
SELECT median2(val) FROM intvals2;

-- Text values
CREATE TABLE textvals2(val text, color int);

INSERT INTO textvals2 VALUES
       ('erik', 1),
       ('mat', 3),
       ('rob', 8),
       ('david', 9),
       ('lee', 2);

SELECT * FROM textvals2 ORDER BY val;
SELECT median2(val) FROM textvals2;

-- Test large table with timestamps
CREATE TABLE timestampvals2 (val timestamptz);

INSERT INTO timestampvals2(val)
SELECT TIMESTAMP 'epoch' + (i * INTERVAL '1 second')
FROM generate_series(0, 100000) as T(i);

SELECT median2(val) FROM timestampvals2;
