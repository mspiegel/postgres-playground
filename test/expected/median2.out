CREATE TABLE intvals2(val int, color text);
-- Test empty table
SELECT median2(val) FROM intvals2;
 median2 
---------
        
(1 row)

-- Integers with odd number of values
INSERT INTO intvals2 VALUES
       (1, 'a'),
       (2, 'c'),
       (9, 'b'),
       (7, 'c'),
       (2, 'd'),
       (-3, 'd'),
       (2, 'e');
SELECT * FROM intvals2 ORDER BY val;
 val | color 
-----+-------
  -3 | d
   1 | a
   2 | c
   2 | e
   2 | d
   7 | c
   9 | b
(7 rows)

SELECT median2(val) FROM intvals2;
 median2 
---------
       2
(1 row)

-- Integers with NULLs and even number of values
INSERT INTO intvals2 VALUES
       (99, 'a'),
       (NULL, 'a'),
       (NULL, 'e'),
       (NULL, 'b'),
       (7, 'c'),
       (0, 'd');
SELECT * FROM intvals2 ORDER BY val;
 val | color 
-----+-------
  -3 | d
   0 | d
   1 | a
   2 | e
   2 | d
   2 | c
   7 | c
   7 | c
   9 | b
  99 | a
     | a
     | e
     | b
(13 rows)

SELECT median2(val) FROM intvals2;
 median2 
---------
       2
(1 row)

-- Text values
CREATE TABLE textvals2(val text, color int);
INSERT INTO textvals2 VALUES
       ('erik', 1),
       ('mat', 3),
       ('rob', 8),
       ('david', 9),
       ('lee', 2);
SELECT * FROM textvals2 ORDER BY val;
  val  | color 
-------+-------
 david |     9
 erik  |     1
 lee   |     2
 mat   |     3
 rob   |     8
(5 rows)

SELECT median2(val) FROM textvals2;
 median2 
---------
 lee
(1 row)

-- Test large table with timestamps
CREATE TABLE timestampvals2 (val timestamptz);
INSERT INTO timestampvals2(val)
SELECT TIMESTAMP 'epoch' + (i * INTERVAL '1 second')
FROM generate_series(0, 100000) as T(i);
SELECT median2(val) FROM timestampvals2;
           median2            
------------------------------
 Thu Jan 01 13:53:20 1970 PST
(1 row)

